src/transaction.rs

```diff
@@ -1440,5 +1440,55 @@ mod tests {
             }
         );
     }
+
+    #[test]
+    fn txout_null_data() {
+        // Output with high opcodes should not be considered nulldata
+        let output: TxOut = hex_deserialize!("\
+            c3319c0000000000d3d3d3d3d3d3d3d3d3d3d3d3fdfdfd0101010101010101010\
+            101010101010101010101010101012e010101010101010101fdfdfdfdfdfdfdfd\
+            fdfdfdfdfdfdfdfdfdfdfdfd006a209f6a6a6a6a6a6a806a6afdfdfdfd17fdfdf\
+            dfdfdfdfdfdfdfdddfdfdfdfdfdfdfdfdfddedededededededededededededede\
+            dedededededededededededededededededededededededededededededededed\
+            edededededededededededededededededededededededededededededededede\
+            dededededea7dedededededededededededededededededfdedededededededed\
+            edededededededede9edededede00000000000001000000000000050000ff0000\
+            000000000000000000ff000000000000000000000000200000000000011c00000\
+            000d3d3d3d3d3d3d3d3d3d3d3d3d3d3d3d3d3d3f3d3d3d3d3d3d3d3d3d3d3d3d3\
+            d3d3d3d3d3d3\
+        ");
+
+        assert!(!output.is_null_data());
+        assert!(!output.is_pegout());
+
+        // Output with pushes that are e.g. OP_1 are nulldata but not pegouts
```

Comment:
I don't see an OP_1 in the hex.

Reply:
Oh, oops, it's actually pushing a literal 0x01. Good call, this is not what I meant to test.